sui-execution/latest/sui-move-natives/src/crypto/sui_state_proof.rs

```diff
@@ -0,0 +1,387 @@
+ *   gas cost: sui_state_proof_verify_link_cap_base   | base cost for function call and fixed opers
+ **************************************************************************************************/
+ 
+ pub fn sui_state_proof_verify_link_cap(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 6);
+
+    // Load the cost parameters from the protocol config
+    let sui_state_proof_cost_params = &context
+        .extensions()
+        .get::<NativesCostTable>()
+        .sui_state_proof_cost_params
+        .clone();
+
+    // Charge the base cost for this oper
+    native_charge_gas_early_exit!(
+        context,
+        sui_state_proof_cost_params.sui_state_proof_verify_link_cap_base
+    );
+
+    let cost = context.gas_used();
+
+
+    let package_id_bytes = pop_arg!(args, Vec<u8>);
+    let type_layout_bytes = pop_arg!(args, Vec<u8>);
+    let transaction_bytes = pop_arg!(args, Vec<u8>);
+    let checkpoint_contents_bytes = pop_arg!(args, Vec<u8>);
+    let summary_bytes = pop_arg!(args, Vec<u8>);
+    let committee_bytes = pop_arg!(args, Vec<u8>);
+
+    let Ok(committee) = bcs::from_bytes::<Committee>(&committee_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_COMMITTEE
+        ));
+    };
+
+    let Ok(summary) = bcs::from_bytes::<CertifiedCheckpointSummary>(&summary_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_CHECKPOINT_SUMMARY
+        ));
+    };
+
+    let Ok(checkpoint_contents) = bcs::from_bytes::<CheckpointContents>(&checkpoint_contents_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_INPUT
+        ));
+    };
+
+    let Ok(transaction) = bcs::from_bytes::<CheckpointTransaction>(&transaction_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_INPUT
+        ));
+    };
+
+
+    let Ok(type_layout) = bcs::from_bytes::<MoveTypeLayout>(&type_layout_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_INPUT
+        ));
+    };
+
+
+    let Ok(package_id_target) = bcs::from_bytes::<ObjectID>(&package_id_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_INPUT
+        ));
+    };
+    
+    // Verify the checkpoint summary using the committee
+    let res = summary.verify_with_contents(&committee, Some(&checkpoint_contents));
+    if let Err(_) = res {
+        return Ok(NativeResult::err(cost, INVALID_TX));
+    }
+
+
+    // Ensure the tx is part of the checkpoint
+    let is_valid_checkpoint_tx = checkpoint_contents.iter().any(|&digest| digest == transaction.effects.execution_digests());
+    if !is_valid_checkpoint_tx {
```

Comment:
```suggestion
    if !checkpoint_contents.iter().any(|&digest| digest == transaction.effects.execution_digests()) {
```
@zeevmoney which why would you prefer?